### PR TITLE
Implement Pull #3043 from citra: appveyor: Determine dlls to include in package programmatically

### PR DESCRIPTION
**Pull request by N00byKing**

### CODE REVIEW

**Comment by FearlessTobi** [comment]
Please squash your commits

**Comment by N00byKing** [comment]
If I understand correctly, squashing can be done automatically on merge (Am not sure though)

File: appveyor.yml [hunk 1]
```diff
@@ -114,23 +114,16 @@ after_build:
...
-          # hardcoded list because we don't build static and determining the list of dlls from the binary is a pain.
-          $MingwDLLs = "Qt5Core.dll","Qt5Widgets.dll","Qt5Gui.dll","Qt5OpenGL.dll",
-                       # QT dll dependencies
-                       "libbz2-*.dll","libicudt*.dll","libicuin*.dll","libicuuc*.dll","libffi-*.dll",
-                       "libfreetype-*.dll","libglib-*.dll","libgobject-*.dll","libgraphite2.dll","libiconv-*.dll",
-                       "libharfbuzz-*.dll","libintl-*.dll","libpcre-*.dll","libpcre2-16-*.dll","libpcre16-*.dll","libpng16-*.dll",
-                       # Runtime/Other dependencies
-                       "libgcc_s_seh-*.dll","libstdc++-*.dll","libwinpthread-*.dll","SDL2.dll","zlib1.dll"
+          . "./.appveyor/UtilityFunctions.ps1"
+          $DLLSearchPath = "C:\projects\yuzu\mingw_build\externals\curl-7_55_1\lib;C:\msys64\mingw64\bin;$env:PATH"
```
Comment (jroweboy) [inline appveyor.yml]: yuzu doesn't have curl so its not needed to have the search path include that in there

`$DLLSearchPath = "C:\msys64\mingw64\bin;$env:PATH"`

File: appveyor.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ before_build:
           # redirect stderr and change the exit code to prevent powershell from cancelling the build if cmake prints a warning
           cmd /C 'cmake -G "Visual Studio 15 2017 Win64" -DYUZU_USE_BUNDLED_QT=1 -DYUZU_USE_BUNDLED_SDL2=1 -DYUZU_USE_BUNDLED_UNICORN=1 .. 2>&1 && exit 0'
         } else {
-          C:\msys64\usr\bin\bash.exe -lc "cmake -G 'MSYS Makefiles' -DYUZU_BUILD_UNICORN=1 -DCMAKE_BUILD_TYPE=Release .. 2>&1"
+          C:\msys64\usr\bin\bash.exe -lc "cmake -G 'MSYS Makefiles' -DUSE_SYSTEM_CURL=1 -DYUZU_BUILD_UNICORN=1 -DCMAKE_BUILD_TYPE=Release .. 2>&1"
```
Comment (jroweboy) [inline appveyor.yml]: yuzu doesn't have curl, remove the `-DUSE_SYSTEM_CURL=1`

File: externals/CMakeLists.txt [hunk 3]
```diff
@@ -49,3 +49,14 @@ if (ARCHITECTURE_x86_64)
     target_include_directories(xbyak INTERFACE ./xbyak/xbyak)
     target_compile_definitions(xbyak INTERFACE XBYAK_NO_OP_NAMES)
 endif()
+
+# curl (if building for Windows)
+if (WIN32)
```
Comment (jroweboy) [inline externals/CMakeLists.txt]: yuzu doesn't have curl so we don't need any changes to this file.

**Comment by N00byKing** [comment]
Shit the curl stuff was from a custom build of mine, will remove

**Comment by N00byKing** [comment]
So the MSVC build finished, artifacts look good

**Comment by N00byKing** [comment]
So the MinGW build finished, while it runs, it's bloated AF (Double the size, too many DLLs)

Something is really wrong with the MinGW build, but I am unsure whether or not this Pull Request is the issue, Can someone weigh in here?

**Comment by jroweboy** [comment]
@N00byKing its not too many dlls, thats the right amount of dlls.

@MerryMage o.O Sorry to drag you into this, and I guess it has something to do with unicorn, but your script added 2to3 https://docs.python.org/2/library/2to3.html to the release as well. Its so small its not a problem to leave it in there, but its weird shipping that with the rest of the emu.

**Comment by N00byKing** [comment]
Oh right, my fault. Forgot the Redistributables

**Comment by MerryMage** [comment]
@jroweboy That seems a bit weird. I'll take a look.

**Comment by Hexagon12** [comment]
What's the status of this PR?

**Comment by N00byKing** [comment]
@Hexagon12 dunno people just forgot it i guess. Apart from the 2to3 stuff it should be ready though

**Comment by jroweboy** [comment]
did anyone look into the issue where the script also pulled in 2to3? i don't think it is bad enough to keep this from shipping, but its a little troubling.

**Comment by N00byKing** [comment]
@jroweboy It is very curious, but it still happens, and I dont know why

**Comment by bunnei** [comment]
@jroweboy please signoff if you think this change is fine - I'll defer to your judgement. I'm not really an expert on our buildsystem/infrastructure/deps maintenance.

**Comment by bunnei** [comment]
`jroweboy - Today at 12:00 PM
@bunnei I'd say just merge it. The issue with 2to3 will likely go away when we toss out unicorn once and for all`

That's a LGTM to me

**Pull request merged**